palette/src/yuv/yuv.rs

```diff
@@ -0,0 +1,203 @@
+/// YUV is an alternate representation for an RGB color space with a focus on separating luminance
+/// from chroma components.
+#[derive(Debug, PartialEq, FromColorUnclamped, Pixel)]
+#[cfg_attr(feature = "serializing", derive(Serialize, Deserialize))]
+#[palette(
+    palette_internal,
+    white_point = "<S::RgbSpace as RgbSpace>::WhitePoint",
+    component = "T",
+    skip_derives(Luma, Rgb)
+)]
+#[repr(C)]
+pub struct Yuv<S: YuvStandard, T: Float = f32> {
+    /// The lumnance signal where `0.0f` is no light and `1.0f` means a maximum displayable amount
+    /// of all colors.
+    pub luminance: T,
+
+    /// The difference between overall brightness and the blue signal. This is centered around
+    /// `0.0f` but its maximum absolute value depends on the standard.
+    pub blue_diff: T,
+
+    /// The difference between overall brightness and the red signal. This is centered around
+    /// `0.0f` but its maximum absolute value depends on the standard.
+    pub red_diff: T,
+
+    /// The kind of YUV standard.
+    #[cfg_attr(feature = "serializing", serde(skip))]
+    #[palette(unsafe_zero_sized)]
+    pub standard: PhantomData<S>,
+}
+
+impl<S: YuvStandard, T: Float> Copy for Yuv<S, T> {}
+
+impl<S: YuvStandard, T: Float> Clone for Yuv<S, T> {
+    fn clone(&self) -> Yuv<S, T> {
+        *self
+    }
+}
+
+impl<S: YuvStandard, T: FloatComponent> Yuv<S, T> {
+    /// Create a YUV color (in YCbCr order).
+    pub fn new(luminance: T, blue_diff: T, red_diff: T) -> Yuv<S, T> {
+        Yuv {
+            luminance,
+            red_diff,
+            blue_diff,
+            standard: PhantomData,
+        }
+    }
+
+    /// Convert into another component type.
+    pub fn into_format<U>(self) -> Yuv<S, U>
+    where
+        U: FloatComponent + FromComponent<T>,
+    {
+        Yuv {
+            luminance: U::from_component(self.luminance),
+            red_diff: U::from_component(self.red_diff),
+            blue_diff: U::from_component(self.blue_diff),
+            standard: PhantomData,
+        }
+    }
+
+    /// Convert from another component type.
+    pub fn from_format<U>(color: Yuv<S, U>) -> Self
+    where
+        T: FromComponent<U>,
+        U: FloatComponent,
+    {
+        color.into_format()
+    }
+
+    /// Convert to a `(luminance, red_diff, blue_diff)` tuple.
+    pub fn into_components(self) -> (T, T, T) {
+        (self.luminance, self.red_diff, self.blue_diff)
+    }
+
+    /// Convert from a `(luminance, red_diff, blue_diff)` tuple.
+    pub fn from_components((luminance, red_diff, blue_diff): (T, T, T)) -> Self {
+        Self::new(luminance, red_diff, blue_diff)
+    }
+
+    fn from_rgb_internal<Sp>(rgb: Rgb<Linear<Sp>, T>) -> Self
+    where
+        T: FloatComponent,
+        Sp: RgbSpace<WhitePoint = <S::RgbSpace as RgbSpace>::WhitePoint>,
+    {
+        let xyz = Xyz::<<S::RgbSpace as RgbSpace>::WhitePoint, T>::from_color(rgb);
+        let rgb = Rgb::<(S::RgbSpace, S::TransferFn), T>::from_color(xyz);
```

Comment:
You should probably use `from_color_unclamped` internally, unless clamping is always required. If it is required, it would be nice to have a comment with a short explanation. Just so it's not "fixed" in the future. The reason why is that certain applications want numbers outside the typical range to not get lost. That's why clamping only happens at the end of the conversion chain for other color spaces.